crates/kitsune_p2p/kitsune_p2p/src/gossip/sharded_gossip/ops.rs

```diff
@@ -308,38 +296,45 @@ impl ShardedGossipLocal {
         };
 
         // Chunk the ops into multiple gossip messages if needed.
-        into_chunks(&mut gossip, missing_ops, complete);
+        into_chunks(&mut gossip, missing_hashes, complete);
 
         Ok(gossip)
     }
 
     /// Incoming ops that were missing from this nodes bloom filter.
-    pub(super) async fn incoming_missing_ops(&self, ops: Vec<KOp>) -> KitsuneResult<()> {
-        // Put the ops in the agents that contain the ops within their arcs.
-        store::put_ops(&self.evt_sender, &self.space, ops).await?;
-
+    pub(super) async fn incoming_missing_ops(
+        &self,
+        source: FetchSource,
+        ops: Vec<KOpHash>,
+    ) -> KitsuneResult<()> {
+        for op_hash in ops {
+            let request = FetchRequest {
+                key: FetchKey::Op { op_hash },
+                author: None,
+                options: None,
+                context: None,
+            };
+            self.fetch_queue.push(request, self.space.clone(), source);
+        }
         Ok(())
     }
 }
 
 /// Separate gossip into chunks to keep messages under the max size.
 // pair(maackle, freesig): can use this for chunking, see above fn for use
-fn into_chunks(gossip: &mut Vec<ShardedGossipWire>, ops: Vec<KOp>, complete: u8) {
-    let mut chunk = Vec::with_capacity(ops.len());
+fn into_chunks(gossip: &mut Vec<ShardedGossipWire>, hashes: Vec<KOpHash>, complete: u8) {
+    let mut chunk = Vec::with_capacity(hashes.len());
     let mut size = 0;
 
     // If there are no ops missing we send back an empty final chunk
     // so the other side knows we're done.
-    if ops.is_empty() {
-        gossip.push(ShardedGossipWire::missing_ops(
-            Vec::with_capacity(0),
-            complete,
-        ));
+    if hashes.is_empty() {
+        gossip.push(ShardedGossipWire::missing_op_hashes(vec![], complete));
     }
 
-    for op in ops {
+    for op in hashes {
         // Bytes for this op.
-        let bytes = op.size();
+        let bytes = op.0.len();
```

Comment:
mmm... so is it using `MAX_SEND_BUF_BYTES` still... just using the hash sizes instead of the full op sizes?

Reply:
Yeah, it seems unnecessary to do batching now, but it was easier to leave the mechanism in than to take it out..